buildings_tools/src/buildings_tools/DrawBuildingAction.java

```diff
@@ -300,7 +324,12 @@ private void drawingAdvance(MouseEvent e) {
 
     private void drawingFinish() {
         if (building.getLength() != 0) {
-            Way w = building.create();
+            Way w;
+            if (ToolSettings.Shape.RECTANGLE.equals(ToolSettings.getShape())) {
+                w = building.createRectangle();
+            } else {
+                w = building.createCircle();
+            }
```

Comment:
I always feel like default behavior (in this case Rectangle) should be in the `else`.